ui/components/app/name/name-details/name-details.tsx

```diff
@@ -54,19 +52,30 @@ import {
 } from '../../../../store/actions';
 import { useCopyToClipboard } from '../../../../hooks/useCopyToClipboard';
 import { useName } from '../../../../hooks/useName';
-import { I18nContext } from '../../../../contexts/i18n';
 import { useDisplayName } from '../../../../hooks/useDisplayName';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
 import { usePetnamesMetrics } from './metrics';
 
 const UPDATE_DELAY = 1000 * 2; // 2 Seconds
 
+// The following lines avoid i18n "unused messages" lint error.
+// t('nameProvider_ens');
+// t('nameProvider_etherscan');
+// t('nameProvider_lens');
+// t('nameProvider_token');
+const getProviderLabel = (sourceId: string) => `nameProvider_${sourceId}`;
```

Comment:
Do we still need the fallback to the name source labels and the source ID in the case of snaps which won't have translation keys?

Worth nothing that I think `t()` throws if a key isn't found so may need a `try catch`.

Reply:
Moved this function below. 
Since we need to make an exception for each translation key, the providers with localized names are known, which avoids the need for a try/catch.